amd_openvx/openvx/ago/ago_haf_cpu_generic_functions.cpp

```diff
@@ -0,0 +1,878 @@
+            vx_int32 src1 = *(vx_uint8 *)src1p;
+            vx_int32 result = (vx_int32)((1 - alpha) * (vx_float32)src1 + alpha * (vx_float32)src0);
+            *(vx_uint8 *)dstp = (vx_uint8)result;
+        }
+    }
+    status |= vxUnmapImagePatch(img1, src_map_id[0]);
+    status |= vxUnmapImagePatch(img2, src_map_id[1]);
+    status |= vxUnmapImagePatch(output, dst_map_id);
+    return status;
+}
+
+// helpers
+static int vx_uint8_compare(const void *p1, const void *p2)
+{
+    vx_uint8 a = *(vx_uint8 *)p1;
+    vx_uint8 b = *(vx_uint8 *)p2;
+    if (a > b)
+        return 1;
+    else if (a == b)
+        return 0;
+    else
+        return -1;
+}
+
+static vx_uint32 readMaskedRectangle(const void *base,
+    const vx_imagepatch_addressing_t *addr,
+    const vx_border_t *borders,
+    vx_df_image type,
+    vx_uint32 center_x,
+    vx_uint32 center_y,
+    vx_uint32 left,
+    vx_uint32 top,
+    vx_uint32 right,
+    vx_uint32 bottom,
+    vx_uint8 *mask,
+    vx_uint8 *destination,
+    vx_uint32 border_x_start)
+{
+    vx_int32 width = (vx_int32)addr->dim_x, height = (vx_int32)addr->dim_y;
+    vx_int32 stride_y = addr->stride_y;
+    vx_int32 stride_x = addr->stride_x;
+    vx_uint16 stride_x_bits = addr->stride_x_bits;
+    const vx_uint8 *ptr = (const vx_uint8 *)base;
+    vx_int32 ky, kx;
+    vx_uint32 mask_index = 0;
+    vx_uint32 dest_index = 0;
+
+    // kx, ky - kernel x and y
+    if (borders->mode == VX_BORDER_REPLICATE || borders->mode == VX_BORDER_UNDEFINED)
+    {
+        for (ky = -(int32_t)top; ky <= (int32_t)bottom; ++ky)
+        {
+            vx_int32 y = (vx_int32)(center_y + ky);
+            y = y < 0 ? 0 : y >= height ? height - 1 : y;
+
+            for (kx = -(int32_t)left; kx <= (int32_t)right; ++kx, ++mask_index)
+            {
+                vx_int32 x = (int32_t)(center_x + kx);
+                x = x < (int32_t)border_x_start ? (int32_t)border_x_start : x >= width ? width - 1 : x;
+                if (mask[mask_index])
+                {
+                    if (type == VX_DF_IMAGE_U1)
+                        ((vx_uint8*)destination)[dest_index++] =
+                            ( *(vx_uint8*)(ptr + y*stride_y + (x*stride_x_bits) / 8) & (1 << (x % 8)) ) >> (x % 8);
+                    else    // VX_DF_IMAGE_U8
+                        ((vx_uint8*)destination)[dest_index++] = *(vx_uint8*)(ptr + y*stride_y + x*stride_x);
+                }
+            }
+        }
+    }
+    else if (borders->mode == VX_BORDER_CONSTANT)
+    {
+        vx_pixel_value_t cval = borders->constant_value;
+        for (ky = -(int32_t)top; ky <= (int32_t)bottom; ++ky)
+        {
+            vx_int32 y = (vx_int32)(center_y + ky);
+            int ccase_y = y < 0 || y >= height;
+
+            for (kx = -(int32_t)left; kx <= (int32_t)right; ++kx, ++mask_index)
+            {
+                vx_int32 x = (int32_t)(center_x + kx);
+                int ccase = ccase_y || x < (int32_t)border_x_start || x >= width;
```

Comment:
Codacy found an issue: [The scope of the variable 'ccase' can be reduced.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=6287409)